testing/e2e/Makefile

```diff
@@ -49,10 +49,17 @@ create-cluster:
 .PHONY: run-tests
 run-tests: set-arch
 	@$(call check_defined, AGENT_VERSION, AGENT_VERSION is not defined)
-	AGENT_VERSION=$(AGENT_VERSION) AGENT_BIN_PATH=$(CURDIR)/$(ARTIFACTS_DIR)/elastic-agent-$(AGENT_VERSION)-linux-$(ARCH)/elastic-agent go test -v -timeout 300s -tags=e2e github.com/elastic/elastic-agent/testing/e2e
+	docker run \
+		--rm \
+		-v $(CURDIR)/../..:/mnt/e2e \
+		-w /mnt/e2e \
+		golang:$(GO_VERSION) \
+			sh -c 'AGENT_VERSION=$(AGENT_VERSION) \
+			AGENT_BIN_PATH=/mnt/e2e/testing/e2e/$(ARTIFACTS_DIR)/elastic-agent-$(AGENT_VERSION)-linux-$(ARCH)/elastic-agent \
+			go test -v -timeout 300s -tags=e2e github.com/elastic/elastic-agent/testing/e2e/tests/...'
```

Comment:
`mage` is the system we use, the `Makefile` should really just be a wrapper. We should move this into the magefile.go.

Reply:
++ agreed!  I'd even go as far as removing the Makefile entirely and introducing the relevant targets in our existing `magefile.go`.   I've added a task (in a private repo meta issue) to track this change.